_pages/clients/python.md

```diff
@@ -64,206 +60,164 @@ client = Algorithmia.client("YOUR_API_KEY", "https://mylocalendpoint")
+In this example, we'll upload a file to Algorithmia's Hosted Data storage provider, invoke the "*TODO*" algorithm which will process the file, upload the result, and return a data URI which we can use to download the file.
 
-### Downloading Data from a Data Collection
+### Create a Data Collection
 
-Next check if the file that you just uploaded to data collections exists, and try downloading it to a (new) local file:
+The documentation for "*TODO*" says that it takes a Data URI as input, but we still need a location to store the data. We'll create a new directory, then update its permissions so that it's publicly accessible:
 
 {% highlight python %}
-# Download the file
-if client.file(text_file).exists() is True:
-    localfile = client.file(text_file).getFile()
+from Algorithmia.acl import ReadAcl, AclType
+# Instantiate a DataDirectory object, set your data URI and call create
+img_directory = client.dir("data://YOUR_USERNAME/img_directory")
+# Create your data collection if it does not exist
+if img_directory.exists() is False:
+    img_directory.create()
+# Change permissions on your data collection to public
+img_directory.update_permissions(ReadAcl.public)
 {% endhighlight %}
 
-This copies the file from your data collection and saves it as a file on your local machine, storing the filename in the variable `localfile`.
+Instead of your username you can also use '.my' when calling algorithms. For more information about the '.my' pseudonym check out the [Hosted Data Guide]({{site.baseurl}}/data/hosted).
+{: .notice-info}
 
-Alternately, if you just need the text content of the file to be stored in a variable, you can retrieve the remote file's content without saving the actual file:
+### Upload Data to your Data Collection
+
+Now we're ready to upload an image file for processing. We'll use *TODO* something.jpg. First create a variable that holds the path in your data collection to the image file you will be uploading:
 
 {% highlight python %}
-# Download contents of file as a string
-if client.file(text_file).exists() is True:
-    input = client.file(text_file).getString()
+img_file = "data://YOUR_USERNAME/img_directory/something.jpg"
 {% endhighlight %}
 
-This will get your file as a string, saving it to the variable `input`.  If the file was binary (an image, etc), you could instead use the function `.getBytes()` to retrieve the file's content as a byte array. For more image-manipulation tutorials, see the [Computer Vision Recipes]({{site.baseurl}}/tutorials/recipes/#computer-vision).
-
-Now you've seen how to upload a local data file, check if a file exists in a data collection, and download the file contents.
-
-For more methods on how to get a file using the Data API from a data collection go to the [API Specification](http://docs.algorithmia.com/#getting-a-file).
-
-{% if site.enterprise %}
```

Comment:
Insights section looks like it was removed in new version - this section needs to appear in the new version as well w/enterprise flag